libs/ha/ha_common_libs_k8s.py

```diff
@@ -1480,3 +1480,106 @@ def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
+            if key == field:
+                fields_found.append(value)
+                if replace_key:
+                    search_dict[replace_key] = search_dict.pop(key)
+                    search_dict[replace_key] = replace_val
+            elif isinstance(value, dict):
+                results = self.get_replace_recursively(value, field, replace_key, replace_val)
+                for result in results:
+                    fields_found.append(result)
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        more_results = self.get_replace_recursively(item, field, replace_key,
+                                                                    replace_val)
+                        for another_result in more_results:
+                            fields_found.append(another_result)
+
+        return fields_found
+
+    def update_deployment_yaml(self, pod_obj, pod_name, find_key, replace_key=None,
+                               replace_val=None):
+        """
+        Function to find and replace key from deployment yaml file
+        :param pod_obj: Object of pod
+        :param pod_name: Name of the pod
+        :param find_key: Key to be searched
+        :param replace_key: Key with which older key to be replaced
+        :param replace_val: Value with which older value to be replaced
+        :return: bool, str, str (status, new path, backup path)
+        """
+        resp = pod_obj.get_deploy_replicaset(pod_name)
+        deploy = resp[1]
+        LOGGER.info("Deployment for pod %s is %s", pod_name, deploy)
+        LOGGER.info("Taking deployment backup")
+        resp = pod_obj.backup_deployment(deploy)
+        yaml_path = resp[1]
+        LOGGER.info("Copy deployment yaml file to local system")
+        pod_obj.copy_file_to_local(remote_path=yaml_path, local_path=yaml_path)
+        resp = config_utils.read_yaml(yaml_path)
+        if not resp[0]:
+            return resp
+        data = resp[1]
+        system_utils.remove_file(yaml_path)
+        LOGGER.info("Update %s of deployment yaml with %s : %s", find_key, replace_key, replace_val)
+        val = self.get_replace_recursively(data, find_key, replace_key, replace_val)
+        if not val:
+            return False, f"Failed to find key {find_key} in deployment yaml {yaml_path} of pod " \
+                          f"{pod_name}"
+        resp = config_utils.write_yaml(fpath=yaml_path, write_data=data, backup=True,
+                                       sort_keys=False)
+        LOGGER.debug("Response: %s", resp)
+        if not resp[0]:
+            return resp
+        LOGGER.info("Copying files %s and %s to master node", yaml_path, f'{yaml_path}.bkp')
+        resp = pod_obj.copy_file_to_remote(local_path=yaml_path, remote_path=yaml_path)
+        if not resp[0]:
+            return resp
+        resp = pod_obj.copy_file_to_remote(local_path=f'{yaml_path}.bkp',
```

Comment:
why do we need to copy backup file as well? is it going to chnage?

Reply:
We should recover deployment for control pod right?
Backup file will have original parameters in it, so that we can restore control pod to its original state.

Reply:
but aren't we taking back already at node ?

Reply:
Backup is taken in yaml_path, and the same is being used for modified yaml. 
As anyway write_yaml creates backup file, i used same path for modified yaml.

Reply:
Agree, if copy fails, the whole test will fail.
Updated accordingly.